src/main/java/uk/gov/hmcts/cmc/claimstore/services/notifications/ResponseNeededNotificationService.java

```diff
@@ -0,0 +1,110 @@
+
+import com.google.common.collect.ImmutableMap;
+import org.quartz.JobDataMap;
+import org.quartz.JobDetail;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.retry.annotation.Backoff;
+import org.springframework.retry.annotation.Recover;
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.cmc.claimstore.appinsights.AppInsights;
+import uk.gov.hmcts.cmc.claimstore.appinsights.AppInsightsEvent;
+import uk.gov.hmcts.cmc.claimstore.config.properties.notifications.EmailTemplates;
+import uk.gov.hmcts.cmc.claimstore.config.properties.notifications.NotificationTemplates;
+import uk.gov.hmcts.cmc.claimstore.config.properties.notifications.NotificationsProperties;
+import uk.gov.hmcts.cmc.claimstore.services.ClaimService;
+import uk.gov.hmcts.cmc.claimstore.utils.Formatting;
+import uk.gov.hmcts.cmc.domain.exceptions.NotificationException;
+import uk.gov.hmcts.cmc.domain.models.Claim;
+import uk.gov.service.notify.NotificationClient;
+import uk.gov.service.notify.NotificationClientException;
+
+import java.time.LocalDate;
+import java.util.Map;
+import java.util.Optional;
+
+import static uk.gov.hmcts.cmc.claimstore.services.notifications.content.NotificationTemplateParameters.CLAIMANT_NAME;
+import static uk.gov.hmcts.cmc.claimstore.services.notifications.content.NotificationTemplateParameters.CLAIM_REFERENCE_NUMBER;
+import static uk.gov.hmcts.cmc.claimstore.services.notifications.content.NotificationTemplateParameters.DEFENDANT_NAME;
+import static uk.gov.hmcts.cmc.claimstore.services.notifications.content.NotificationTemplateParameters.FRONTEND_BASE_URL;
+import static uk.gov.hmcts.cmc.claimstore.services.notifications.content.NotificationTemplateParameters.RESPOND_TO_CLAIM_URL;
+import static uk.gov.hmcts.cmc.claimstore.services.notifications.content.NotificationTemplateParameters.RESPONSE_DEADLINE;
+
+@Service
+public class ResponseNeededNotificationService {
+    private final Logger logger = LoggerFactory.getLogger(ResponseNeededNotificationService.class);
+
+    private final NotificationClient notificationClient;
+    private final NotificationsProperties notificationsProperties;
+    private final AppInsights appInsights;
+    private final ClaimService claimService;
+
+    @Autowired
+    public ResponseNeededNotificationService(
+        NotificationClient notificationClient,
+        NotificationsProperties notificationsProperties,
+        AppInsights appInsights,
+        ClaimService claimService) {
+        this.notificationClient = notificationClient;
+        this.notificationsProperties = notificationsProperties;
+        this.appInsights = appInsights;
+        this.claimService = claimService;
+    }
+
+    @Retryable(value = NotificationException.class, backoff = @Backoff(delay = 200))
+    public void sendMail(JobDetail jobDetail) {
```

Comment:
I feel more like `uk.gov.hmcts.cmc.claimstore.jobs.NotificationEmailJob#execute` should be responsible of checking whether notification can be send and `uk.gov.hmcts.cmc.claimstore.services.notifications.ResponseNeededNotificationService#sendMail` just send email.

@dharmendrak consider effort needed to change it. If too many I am happy to leave it as is for now. I know you have quite a few test cases that would need a change as well.

SUGESSTION ONLY

Reply:
how about a retest after changes @geek-soft - seems we have a preview env (from the view deployment link below)

Reply:
changes are cosmetic @pragnesh-patel

Reply:
pushed in, pull latest